8_reference/70_techniques.txt

```diff
@@ -0,0 +1,309 @@
+
+===== Variables definitions in the <SECTION> tags
+
+There are three tags to create a variable:
+
+   * SELECT1: Can select only one value out of several. If there are less than 3 possible values, displays radio buttons, otherwise a select field.
+   * SELECT: Can select several values out of al the possibles. Displays checkboxes.
+   * INPUT: Displays an input field (that can be tuned)
+
+----
+
+<SELECT1/SELECT/INPUT>                                                        <!-- Depend on the display and bvehaviour needed -->
+  <NAME>variableName</NAME>
+  <DESCRIPTION>variableDescription</DESCRIPTION>
+  <LONGDESCRIPTION>longDescription</LONGDESCRIPTION>                          <!-- Optional, set the text in the tooltips -->
+  <UNIQUEVARIABLE>true/false</UNIQUEVARIABLE>                                 <!-- Optional, default false; if true, this variable will have the same value over all the instance of this template for a given node -->
+  <ITEM>                                                                      <!-- Only for SELECT and SELECT1, list of selectable values -->
+    <VALUE>value</VALUE>                                                      <!-- value that will be put in the template-->
+    <LABEL>humanReadableText</LABEL>                                          <!-- value displayed in the web interface --> 
+  </ITEM>
+  <CONSTRAINT>                                                                <!-- Optional, defines some constraints on values -->
+    <DEFAULT>defaultValue</DEFAULT>                                           <!-- Optional; Defines a default value -->
+    <TYPE>variableType</TYPE>                                                 <!-- Optional; default string; variable type -->
+    <MAYBEEMPTY>true/false</MAYBEEMPTY>                                       <!-- Optional; default false; defines if the variable is optional or not; only for the INPUT variable -->
+    <REGEX error="errorMsg">regex</REGEX>                                     <!-- Optional; only for the INPUT variable; efine a regular expression the variable should match, and an optional error message -->
+    <PASSWORDHASH>hashtype</PASSWORDHASH>                                     <!-- Optional; only for the password TYPE variable; define the way a password will be handled (hashed or not, hash types allowed ...) -->
+  </CONSTRAINT>
+</SELECT1/SELECT/INPUT>
+
+----
+
+Note: It is possible to inline LABEL and VALUE in the ITEM tag
+
+----
+
+<ITEM label="Red" value="red"/>
+
+----
+
+is equivalent to
+
+----
+
+<ITEM>
+ <LABEL>Red</LABEL>
+ <VALUE>red</VALUE>
+</ITEM>
+
+----
+
+NOTE: INPUT fields in pre-2.6 needed to be manually escaped. Indeed, the CFEngine syntax doesn't allow quotes ( " ) within string, and the user had to manually escape them. Since version 2.6 of Rudder, the defaults INPUT fields are automatically escaped, meaning any quote will be written in the policies as \" ; and any backslash will be written as \\
```

Comment:
I think we can drop the reference to 2.6, and say:
"NOTE: INPUT fields are automatically escaped, meaning any quote will be ...